cmd/nitro/nitro.go

```diff
@@ -309,15 +309,38 @@ func mainImpl() int {
 
 	var l1TransactionOpts *bind.TransactOpts
 	var dataSigner signature.DataSignerFunc
+	var l1TransactionOptsValidator *bind.TransactOpts
+	var l1TransactionOptsBatchPoster *bind.TransactOpts
 	sequencerNeedsKey := nodeConfig.Node.Sequencer.Enable && !nodeConfig.Node.Feed.Output.DisableSigning
 	setupNeedsKey := l1Wallet.OnlyCreateKey || nodeConfig.Node.Staker.OnlyCreateWalletContract
 	validatorCanAct := nodeConfig.Node.Staker.Enable && !strings.EqualFold(nodeConfig.Node.Staker.Strategy, "watchtower")
+	if *l1Wallet != genericconf.WalletConfigDefault && nodeConfig.Node.Staker.L1Wallet != genericconf.WalletConfigDefault && nodeConfig.Node.BatchPoster.L1Wallet != genericconf.WalletConfigDefault {
+		log.Crit("--l1.l1-wallet, --node.staker.l1-wallet and --node.batch-poster.l1-wallet all cannot be set at the same time")
+	}
 	if sequencerNeedsKey || nodeConfig.Node.BatchPoster.Enable || setupNeedsKey || validatorCanAct {
 		l1TransactionOpts, dataSigner, err = util.OpenWallet("l1", l1Wallet, new(big.Int).SetUint64(nodeConfig.L1.ChainID))
```

Comment:
Does this still require `l1.l1-wallet` to be set if a wallet is required. Seems like this will fail otherwise

Reply:
fixed.